### ISSUE TITLE
Check FPS

### ISSUE THREAD

**Issue by AbhishekPSI7042** [ultralytics/yolov5#12414]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I want check FPS also while running the detect.py. So, I have added this code in my detect.py is it right or not please tell me.

# Stream results
im0 = annotator.result()
fps_text = f"FPS: {1 / dt[1].dt:.2f}"
cv2.putText(im0, fps_text, (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 255, 0), 2, cv2.LINE_AA)

**Comment by glenn-jocher** [comment]
@AbhishekPSI7042 you can calculate FPS by adding the following code to your `detect.py` after `im0 = annotator.result()`:

```python
fps_text = f"FPS: {1 / dt[1].dt:.2f}"
cv2.putText(im0, fps_text, (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 255, 0), 2, cv2.LINE_AA)
```

Your implementation looks correct! This will display the FPS on the output image. Remember to import `cv2` if you haven't already. For more details, refer to the documentation at https://docs.ultralytics.com/yolov5/. Keep up the great work!

**Comment by AbhishekPSI7042** [comment]
@glenn-jocher hey, thanks I have another question I wan to save my model into fp16 in pytorch format how can do that. And can we improve the fps on cpu currently I'm getting 27fps on my cpu with image size 360 is there any other way to increase fps?

**Comment by glenn-jocher** [comment]
@AbhishekPSI7042 you can save the model in fp16 format using PyTorch's `torch.save` function with `torch.half` dtype. To increase FPS on CPU, try reducing the image size or using a smaller model variant like YOLOv5s. Keep in mind, increasing FPS on CPU may sacrifice accuracy or vice versa. Always refer to the YOLOv5 documentation at https://docs.ultralytics.com/yolov5/ for best practices. Good luck!

**Comment by AbhishekPSI7042** [comment]
@glenn-jocher how can I save the model in pytorch because when I use export.py there is no way to save model in .pt format in fp16.

**Comment by glenn-jocher** [comment]
@AbhishekPSI7042 To save the model in PyTorch's .pt format with fp16 precision, you can directly use PyTorch's `torch.save` function and specify the dtype as `torch.half` for the model's state dict. For example:

```python
model.half()  # Convert model to half precision
torch.save(model.state_dict(), 'model_fp16.pt')  # Save model with fp16 precision
```

This will save the model with fp16 precision in .pt format. For more details, refer to the YOLOv5 documentation at https://docs.ultralytics.com/yolov5/. Keep exploring and feel free to reach out with any more questions!
